src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -164,7 +170,7 @@ void analyzeMetricsData(CodeAnalysisMetricData codeAnalysisMetricData)
                             var arg1 = symbol.Name;
                             var arg2 = codeAnalysisMetricData.DepthOfInheritance;
                             var arg3 = inheritanceThreshold + 1;
-                            var arg4 = string.Join(", ", ((INamedTypeSymbol)symbol).GetBaseTypes().Select(t => t.Name));
+                            var arg4 = string.Join(", ", ((INamedTypeSymbol)symbol).GetBaseTypes(t => !inheritanceExcludedTypes.Contains(t)).Select(t => t.Name));
```

Comment:
It might be better to store the base types we have kept directly in the metrics data rather than doing multiple calculations.

Reply:
@mavasani Shall we ignore this point for now? Is it worth adding a ticket?

Reply:
Yes, possibly. I am fine to wait till we see a performance issue.